src/cozmo/world.py

```diff
@@ -611,13 +611,64 @@ def send_available_objects(self):
+            if isinstance(obj, objects.FixedCustomObject):
+                logger.info("Removing FixedCustomObject instance: id %s = obj '%s'", id, obj)
+                del self._objects[id]
+
     async def delete_all_custom_objects(self):
-        """Causes the robot to forget about all custom objects it currently knows about."""
+        """Causes the robot to forget about all custom (fixed + marker) objects it currently knows about.
+        
+        Note: This includes all fixed custom objects, and all custom marker object instances, 
+        BUT this does NOT remove the custom marker object definitions, so Cozmo
+        will continue to add new objects if he sees the markers again. To remove
+        the definitions for those objects use: :meth:`undefine_all_custom_marker_objects`
+        """
         msg = _clad_to_engine_iface.DeleteAllCustomObjects()
         self.conn.send_msg(msg)
-        # TODO: use a filter to wait only for a message for the active robot
         await self.wait_for(_clad._MsgRobotDeletedAllCustomObjects)
-        # TODO: reset local object stte
+        self._remove_custom_marker_object_instances()
+        self._remove_fixed_custom_object_instances()
+
+    async def delete_custom_marker_objects(self):
+        """Causes the robot to forget about all custom marker objects it currently knows about.
+
+        Note: This removes custom marker object instances only, it does NOT remove
+        fixed custom objects, nor does it remove the custom marker object definitions, so Cozmo
+        will continue to add new objects if he sees the markers again. To remove
+        the definitions for those objects use: :meth:`undefine_all_custom_marker_objects`
+        """
+        msg = _clad_to_engine_iface.DeleteCustomMarkerObjects()
+        self.conn.send_msg(msg)
+        await self.wait_for(_clad._MsgRobotDeletedCustomMarkerObjects)
+        self._remove_custom_marker_object_instances()
+
+    async def delete_fixed_custom_objects(self):
+        """Causes the robot to forget about all fixed custom objects it currently knows about.
+
+        Note: This removes fixed custom objects only, it does NOT remove
+        the custom marker object instances or definitions.
+        """
+        msg = _clad_to_engine_iface.DeleteFixedCustomObjects()
+        self.conn.send_msg(msg)
+        await self.wait_for(_clad._MsgRobotDeletedFixedCustomObjects)
+        self._remove_fixed_custom_object_instances()
+
+    async def undefine_all_custom_marker_objects(self):
+        """Remove all custom marker object definitions, and any instances of them in the world."""
+        msg = _clad_to_engine_iface.UndefineAllCustomMarkerObjects()
+        self.conn.send_msg(msg)
+        await self.wait_for(_clad._MsgRobotDeletedCustomMarkerObjects)
+        self._remove_custom_marker_object_instances()
+        # Remove all custom object definitions / archetypes
+        self.custom_objects = {}
```

Comment:
It might be better to do `self.custom_objects.clear()` to modify the existing dict, rather than creating a new one - that way if anyone has assigned it to a local variable for easy reference, it'll continue to refer to the right thing.